Added test for correctly passing deformation gradient to UMAT interface

WulfHans opened the pull request:
Added test for correctly passing deformation gradient to UMAT interface (refs #21797)

- the UMAT interface needs to account for the fact, that C is row-major and Fortran is column-major
- test added, that calls the UMAT interface with a simple shear of F_xy=0.5
- demonstrates issue #21797

WulfHans committed:
Added test for correctly passing deformation gradient to UMAT interface (#21797)

- the UMAT interface needs to account for the fact, that C is row-major and Fortran is column-major
- test added, that calls the UMAT interface with a simple shear of F_xy=0.5
- demonstrates issues #21797

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1140398/) on 16caada wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/21829/site)

This comment will be updated on new commits.

WulfHans committed:
Fixed column vs row major bug in UMAT interface

- refs #21797
- because of difference between row and column major, unsymmetric tensors must be transposed before passing them between C and Fortran
- remark: print_shear_print now fails, because it was ill-designed

WulfHans committed:
Corrrected test umat/print.print_shear_print

- refs #21797
- The test was ill-designed because it accepted the transposed of the correct deformation gradient
- now correct deformation gradient and rotation tensor are expected

dschwen commented:
Ok, go ahead and add your fixes.

jessecarterMOOSE commented:
superseded by #21872
